chia/util/block_cache.py

```diff
@@ -1,109 +1,48 @@
 from __future__ import annotations
 
-import logging
-from typing import Dict, List, Optional
+from typing import Dict, Optional
 
 from chia.consensus.block_record import BlockRecord
-from chia.consensus.blockchain_interface import BlockchainInterface
+from chia.consensus.blockchain_interface import BlockRecordsInterface
 from chia.types.blockchain_format.sized_bytes import bytes32
-from chia.types.blockchain_format.sub_epoch_summary import SubEpochSummary
-from chia.types.header_block import HeaderBlock
-from chia.types.weight_proof import SubEpochChallengeSegment, SubEpochSegments
 from chia.util.ints import uint32
 
 
-class BlockCache(BlockchainInterface):
+class BlockCache(BlockRecordsInterface):
```

Comment:
You don't inherit from a protocol to implement it.  The checking is intended to be implicit.  But..  I prefer something explicit and sadly this is the best (or what I picked as best) form we had last I dug into this.

```
    if TYPE_CHECKING:
        _protocol_check: ClassVar[BlockRecordsInterface] = cast("BlockCache", None)
```

Reply:
oh, I see

Reply:
I was under the impression that mypy will issue warnings if you inherit from a protocol and leave some method unimplemented, which I think is a desirable feature

Reply:
I think you just inherit the underlying methods.

https://mypy-play.net/?mypy=latest&python=3.12&flags=strict&gist=658ea13295674ffbd389c5b0fa972e48
```python
from typing import Protocol, cast


class A(Protocol):
    a: int
    
    def m(self) -> None: ...


class B(A):
    pass
```
No complaints from mypy.

One of the big points for me of using protocols instead of ABCs (which you do inherit from and do complain about not-implemented abstract methods) is to not inherit.